include/uv.h

```diff
@@ -243,6 +243,11 @@ typedef enum {
 UV_EXTERN unsigned int uv_version(void);
 UV_EXTERN const char* uv_version_string(void);
 
+typedef void* (*uv_malloc_cb)(size_t size);
+typedef void (*uv_free_cb)(void* ptr);
+
+UV_EXTERN void uv_replace_allocator(uv_malloc_cb malloc_cb, uv_free_cb free_cb);
```

Comment:
Sorry for being that guy... there are not really callbacks, can we use some other suffix file "func"? That would be uv_malloc_func and uv_free_func.

Another (maybe stupid idea): have a single function: uv_allocator_func, here is an example default implementation:

~~~~
void* uv_default_allocator(void* ptr, size_t size) {
    if (!ptr && !size)
        return NULL;
    if (!ptr && size)
        return malloc(size);
    if (ptr && size)
        return realloc(ptr, size);
    if (ptr && !size) {
        free(ptr);
        return NULL;
    }
}
~~~~

Of course this can be compacted, but you get the idea :-)